tests/zfs-tests/tests/functional/zvol/zvol_misc/zvol_misc_002_pos.ksh

```diff
@@ -104,8 +102,26 @@ block_device_wait
 
 fsck -n ${ZVOL_RDEVDIR}/$TESTPOOL/$TESTVOL@snap >/dev/null 2>&1
 retval=$?
-if [[ $retval -ne 0 && $retval -ne $EXIT_STATUS ]]; then
-	log_fail "fsck exited with wrong value $retval "
+
+fail=0
+if [ $retval -ne 0 ] ; then
+	if [ is_linux ] ; then
+		# Linux's fsck returns a different code for this test depending
+		# on the version:
+		#
+		# e2fsprogs-1.43.3 (Fedora 25 and older): returns 4
+		# e2fsprogs-1.43.4 (Fedora 26): returns 8
+		#
+		# https://github.com/zfsonlinux/zfs/issues/6297
+		if [ $retval -ne 4 -a $retval -ne 8 ] ; then
+			fail=1
+		fi
+	elif [ $retval -ne 39 ] ; then
```

Comment:
nit: I think it would be easier to read if you didn't use an `elif` here. I'd use an `else` then put the `if [ $retval -ne 39 ]` block in there.